### PR TITLE
Use JSchema for validation; remove dependency on VS JSON libs

### PR DESCRIPTION
**Pull request by lgolding**
Also:

* Remove the Sarif.Validator project (a library which wrapped the VS JSON library) and the SarifValidator project (a command-line driver for Sarif.Validator). JSchema now supplies the equivalent of both of those.

@michaelcfanning

### CODE REVIEW

File: src/Sarif.FunctionalTests/SarifValidatorTests.cs [hunk 1]
```diff
@@ -22,38 +23,47 @@ public class SarifValidatorTests
...
 
         [Theory]
         [MemberData(nameof(ConverterTestCases))]
         public void Converter_validation(string inputFile)
         {
-            IEnumerable<JsonError> errors = Validator.ValidateFile(inputFile, JsonSchemaFile);
+            string instanceText = File.ReadAllText(inputFile);
+            var validator = new Validator(_schema);
+
+            string[] errors = validator.Validate(instanceText);
```
Comment (lgolding) [inline src/Sarif.FunctionalTests/SarifValidatorTests.cs]: At present, the JSchema validator just returns an array of formatted error messages. When I modify the command line tool to emit SARIF, I'll probably need the validator to return an array of error objects -- possibly `Result` objects -- similar to what the VS-based validator did. (The VS-based validator returned an array of "`JsonError`" objects which the command-line tool turned into `Result` objects. But there's no reason the JSchema validator couldn't return `Result` objects directly.)

File: src/Tools/SarifValidator/App.config [hunk 2]
```diff
@@ -1,30 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (lgolding) [inline src/Tools/SarifValidator/App.config]: This was the VS-based command line tool.

File: src/Sarif.Validation/JsonError.cs [hunk 3]
```diff
@@ -1,35 +0,0 @@
-﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT
```
Comment (lgolding) [inline src/Sarif.Validation/JsonError.cs]: This was the VS-based validation library.

File: src/Sarif.FunctionalTests/Sarif.FunctionalTests.csproj [hunk 4]
```diff
@@ -18,9 +18,9 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory).., build.props))\build.props" />
   <PropertyGroup>
     <!-- Override the TargetFrameworkVersion specified in build.props. This project must target
-         .NET 4.6 because the Sarif.Validator project, upon which this project depends, targets
+         .NET 4.5.2 because the Microsoft.Json.Schema package, upon which this project depends, targets
```
Comment (michaelcfanning) [inline src/Sarif.FunctionalTests/Sarif.FunctionalTests.csproj]: can't JSON.Schema compile against 4.5.1? The rest of SARIF does. Also, we set this property in build.props, do you don't need TargetFrameworkVersion here at all (except to override it as you've done)

**Comment by michaelcfanning** [comment]
nice! +1 love seeing all the code going away. :)

File: src/Sarif.FunctionalTests/Sarif.FunctionalTests.csproj (hunk 4 above)
Comment (lgolding) [inline src/Sarif.FunctionalTests/Sarif.FunctionalTests.csproj]: JSchema can almost certainly build against 4.5.1; I'll fix it.

I don't understand your second sentence. It seems to recapitulate the comment ("Override the TargetFrameworkVersion...").

**Comment by lgolding** [comment]
Well, the code hasn't exactly gone away; it's just hiding ;-)

**Pull request merged**